oauthenticator/gitlab.py

```diff
@@ -23,10 +24,32 @@
 
 from .oauth2 import OAuthLoginHandler, OAuthenticator
 
+GITLAB_URL = os.getenv('GITLAB_URL')
+GITLAB_HOST = os.getenv('GITLAB_HOST')
+
+if not GITLAB_URL and GITLAB_HOST:
+    warnings.warn('Use of GITLAB_HOST has been deprecated in v0.8. '
+                  'Rename GITLAB_HOST environemnt variable to GITLAB_URL.',
+                  DeprecationWarning)
+    if GITLAB_HOST.startswith('https://') or \
+            GITLAB_HOST.startswith('http://'):
+        GITLAB_URL = GITLAB_HOST
+    else:
+        # Hides common mistage of users which set the GITLAB_HOST
+        # without a protocol specification.
+        GITLAB_URL = 'https://{0}'.format(GITLAB_HOST)
+        warnings.warn('The https:// prefix has been added to GITLAB_HOST.'
+                      'Set GITLAB_URL="{0}" instead.'.format(GITLAB_URL))
+
 # Support gitlab.com and gitlab community edition installations
-GITLAB_HOST = os.environ.get('GITLAB_HOST') or 'https://gitlab.com'
+if not GITLAB_URL:
+    GITLAB_URL = 'https://gitlab.com'
+
+# Use only GITLAB_URL in the code bellow.
+del GITLAB_HOST
```

Comment:
What do others think of this pattern? I've not seen it before. I can see the motivation for it but it still feels weird :)

Reply:
I wanted to make it explicit so the global variable is not used in the code bellow.

Reply:
I do not think that we should perform a destructive action on the user's system, especially without any warning.

Reply:
It's not affecting user's system. It's keeping the global namespace in the python file clean.